packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -354,22 +404,29 @@ export class BlockNoteEditor<
       },
     };
 
-    this._tiptapEditor = new BlockNoteTipTapEditor(
-      tiptapOptions,
-      this.schema.styleSchema
-    ) as BlockNoteTipTapEditor & {
-      contentComponent: any;
-    };
+    if (!this.headless) {
+      this._tiptapEditor = new BlockNoteTipTapEditor(
+        tiptapOptions,
+        this.schema.styleSchema
+      ) as BlockNoteTipTapEditor & {
+        contentComponent: any;
+      };
+      this._pmSchema = this._tiptapEditor.schema;
+    } else {
+      // In headless mode, we don't instantiate an underlying TipTap editor,
+      // but we still need the schema
+      this._pmSchema = getSchema(tiptapOptions.extensions!);
+    }
   }
 
   /**
    * Mount the editor to a parent DOM element. Call mount(undefined) to clean up
    *
-   * @warning Not needed for React, use BlockNoteView to take care of this
+   * @warning Not needed to call manually when using React, use BlockNoteView to take care of mounting
    */
-  public mount(parentElement?: HTMLElement | null) {
+  public mount = (parentElement?: HTMLElement | null) => {
```

Comment:
I guess this change is something to do with using `this`, but could you explain specifically why it's needed?

Reply:
added a separate PR comment for this